R/cube-transforms.R

```diff
@@ -208,21 +203,37 @@ setMethod("transforms<-", c("CrunchCube", "list"), function (x, value) {
             # grab the matching insertions, make sure they are proper Insertions
             # and then add them to the dimensions to return.
             one_trans <- value[[dim_name]]
+            vars <- variables(x)
+            cats <- categories(vars[[which(aliases(vars) == dim_name)]])
             one_trans$insertions <- Insertions(
                 data = lapply(one_trans$insertions, makeInsertion,
-                              var_categories = categories(variables(x)[[dim_name]]))
+                              var_categories = cats)
             )
             dim_out$references$view$transform <- jsonprep(one_trans)
         }
         return(dim_out)
     }))
-    
+
     # rename, replace the dimensions and return the cube
     names(dims) <- dimnames
     dimensions(x) <- dims
     return(invisible(x))
 })
 
+# error iff the names are not a dimension in the cube provided
+namesInDims <- function (names, cube) {
```

Comment:
The name of this function doesn't make it clear enough that this called only for validating the inputs.